packages/amplify-environment-parameters/src/environment-parameter-manager.ts

```diff
@@ -85,6 +85,49 @@ class EnvironmentParameterManager implements IEnvironmentParameterManager {
     return !!this.resourceParamManagers[getResourceKey(category, resource)];
   }
 
+  canBeClonedHeadlessly(): boolean {
+    const categoriesThatCannotBeCloned = [
+      AmplifyCategories.API,
+      AmplifyCategories.AUTH,
+      AmplifyCategories.NOTIFICATIONS,
+      AmplifyCategories.STORAGE,
+    ];
+    const resourceKeys = Object.keys(this.resourceParamManagers);
+    const categoryResourceNamePairs: string[][] = resourceKeys.map((key) => key.split('_'));
+    for (const [category, resourceName] of categoryResourceNamePairs) {
+      if (categoriesThatCannotBeCloned.includes(category)) {
+        const resourceParamManager: ResourceParameterManager = this.getResourceParamManager(category, resourceName);
+        if (resourceParamManager.hasAnyParams()) {
+          return false;
+        }
+      }
+    }
+    const envSecrets = this.getEnvSecrets();
+    for (const category of categoriesThatCannotBeCloned) {
```

Comment:
I would suggest lifting `categoriesThatCannotBeCloned` to a properly of the class. That way `getEnvSecrets` can access the properly and then that method can be changed to `canCloneSecrets` to keep all of this logic together